### ISSUE TITLE
Unknown dependency Yii::$app->settings

### ISSUE THREAD

**Issue by boehmi1988** [dmstr/yii2-filefly-module#33]
The code uses calls to Yii::$app->settings in different places, but Yii::$app->settings is not a native Yii2 component, so it crashes with 
`Exception (Unknown Property) 'yii\base\UnknownPropertyException' with message 'Getting unknown property: yii\web\Application::settings' `

On which lib / extension does the code rely here? can it be added to composer.json ?

**Comment by Quexer69** [comment]
Hi :) 

thanks for reporting. indeed that is a non native component.

`Yii::$app->settings` in this case relies on https://github.com/phemellc/yii2-settings.

As a quick fix you can simply add 

`"pheme/yii2-settings": "*"`

to your composer.json.

@schmunk42 @handcode what do you think ? Should we remove the dependency from code or add the settings module to the project?

Thanks in advance :)

**Comment by schmunk42** [comment]
> @schmunk42 @handcode what do you think ? Should we remove the dependency from code or add the settings module to the project?
> 
> Thanks in advance :)

Yeah, we'll take a look, but will be likely next year or you can come up with a PR, which makes this configurable or "use-when-installed".

The dependency should be in `suggest` in `composer.json` in the end.

CC: @eluhr @handcode @H-Algabri @a-klapper
